src/_guides/language/coming-from/js-to-dart.md

```diff
@@ -0,0 +1,2835 @@
+T transform<T, Q>(T param1, Q param2) {
+   ...
+}
+// Calling the method with explicitly defined types.
+transform<int, String>(5, 'string value');
+// Types are optional when the analyzer can infer them.
+transform(5, 'string value');
+```
+
+### Generic classes
+
+Generics can also be applied to classes.
+You can include the type to use when calling a constructor,
+which allows you to tailor reusable classes to specific types.
+
+In the following example, the `Cache` class is for
+caching specific types:
+
+_Dart_
+```dart
+class Cache<T> {
+  T getByKey(String key) {}
+  void setByKey(String key, T value) {}
+}
+// Creating a cache for strings
+var stringCache = Cache<String>(); // stringCache has type Cache<String>
+stringCache.setByKey('Foo', 'Bar'); // Valid, setting a string value.
+stringCache.setByKey('Baz', 5); // Invalid, int type does not match generic.
+```
+
+If the type declaration is omitted,
+the runtime type will be `Cache<dynamic>`
+and both calls to `setByKey` are valid.
+
+### Restricting generics
+
+You can use generics to restrict your code to
+a family of types using `extends`. This ensures
+that your class is instantiated with a generic type
+that extends a specific type:
+
+_Dart_
+```dart
+class NumberManager<T extends num> {
+   ...
+}
+// Valid.
+var manager = NumberManager<int>(); 
+var manager = NumberManager<double>(); 
+// Invalid, String nor its parent classes extend num.
+var manager = NumberManager<String>(); 
+```
+
+### Generics in literals
+
+`Map`-, `Set`- and `List` literals can explicitly
+declare generic types, which is useful when the
+type isn’t inferred or is incorrectly inferred. 
+
+For example, the `List` class has a generic definition:
+`class List<E>`. Generic type `E` refers to the type of
+the list’s contents. Normally, this type is automatically inferred,
+which is used in some `List` class’s member types.
+(For example, its first getter returns a value of type `E`.)
+When defining a `List` literal,
+you can explicitly define the generic type as follows:
+
+_Dart_
+```dart
+// Automatic type inference
+var objList = [5, 2.0]; // Type: List<num>
+// Explicit type definition:
+var objList = <Object>[5, 2.0]; // Type: List<Object>
+// Sets work identically:
+var objSet = <Object>{5, 2.0};
+```
+
+This is also true for `Map`s,
+which also define their key and value types
+using generics (`class Map<K, V>`):
+
+_Dart_
+```dart
+// Automatic type inference
+var map = {
+  'foo': 'bar'
+}; // Type: Map<String, String>
+// Explicit type definition:
+var map = <String, Object>{
+  'foo': 'bar'
+}; // Type: Map<String, Object>
+```
+
+## Doc comments
+
+Regular comments work the same in Dart as they do
+in JavaScript. Using `//` comments out everything beyond
+it for the remaining line, and you can use `/* ... */`
+to block comments spanning multiple lines.
+
+In addition to regular comments,
+Dart also has [doc comments][] that work in tandem
+with [dartdoc][]: a first party tool that generates
```

Comment:
We've been pretty consistent in referring to the tool as `dart doc` across the website now. Which matches mentions of other tools on the page as well:

```suggestion
with [`dart doc`][]: a first party tool that generates
```